File: aedvent-code-2024/day 01/Toon - Python/part1.py

if __name__ == '__main__':
    file = open('./input.txt')
    lines = file.readlines()
    left, right = [], []

    for line in lines:
        l, r = line.split('   ')
        r=int(r.strip())
        l=int(l)
        left.append(l)
        right.append(r)

    left.sort()
    right.sort()
    total = []
    for i in range(len(left)):
        total.append(abs(left[i] - right[i]))
        
        
    print(sum(total))

File: aedvent-code-2024/day 01/Jeroen - TS/part-1.ts
export default function (blob: string) {
  const input = blob
    .trim()
    .split('\n')
    .map((line) => line.trim().split('  '))

  const b = input.map(([_, b]) => +b).sort((a, b) => a - b)
  return input
    .map(([a, _]) => +a)
    .sort((a, b) => a - b)
    .reduce((sum, value, i) => sum + Math.abs(+value - +b[i]), 0)
}
